pkg/kudoctl/util/k8s/k2o.go

```diff
@@ -0,0 +1,188 @@
+	if err != nil {
+		return false
+	}
+	fmt.Printf("framework.kudo.k8s.io/%s unchanged\n", framework.Name)
+	return true
+}
+
+// AnyFrameworkVersionExistsInCluster checks if any FrameworkVersion object matches to the given Framework name
+// in the cluster
+func (k *K2oClient) AnyFrameworkVersionExistsInCluster(framework string) bool {
+	fv, err := k.clientset.KudoV1alpha1().FrameworkVersions(vars.Namespace).List(v1.ListOptions{})
+	if err != nil {
+		return false
+	}
+	if len(fv.Items) < 1 {
+		return false
+	}
+
+	var i int
+	for _, v := range fv.Items {
+		if strings.HasPrefix(v.Name, framework) {
+			i++
+		}
+	}
+	if i < 1 {
+		return false
+	}
+	fmt.Printf("frameworkversion.kudo.k8s.io/%s unchanged\n", framework)
+	return true
+}
+
+// AnyInstanceExistsInCluster checks if any FrameworkVersion object matches to the given Framework name
+// in the cluster.
+// An Instance has two identifiers:
+// 		1) Spec.FrameworkVersion.Name
+// 		spec:
+//    		frameworkVersion:
+//      		name: kafka-2.11-2.4.0
+// 		2) LabelSelector
+// 		metadata:
+//    		creationTimestamp: "2019-02-28T14:39:20Z"
+//    		generation: 1
+//    		labels:
+//      		controller-tools.k8s.io: "1.0"
+//      		framework: kafka
+// This function also just returns true if the Instance matches a specific FrameworkVersion of a Framework
+func (k *K2oClient) AnyInstanceExistsInCluster(name, version string) bool {
+	instances, err := k.clientset.KudoV1alpha1().Instances(vars.Namespace).List(v1.ListOptions{LabelSelector: "framework=" + name})
+	if err != nil {
+		return false
+	}
+	if len(instances.Items) < 1 {
+		return false
+	}
+
+	// Todo: check function that actual checks for the FrameworkVersion named e.g. "test-1.0" to exist
+	var i int
+	for _, v := range instances.Items {
+		if v.Spec.FrameworkVersion.Name == name+"-"+version {
+			i++
+		}
+	}
+
+	// This is when we don't find the version we are looking for
+	if i < 1 {
+		return false
+	}
+	fmt.Printf("instance.kudo.k8s.io/%s unchanged\n", name)
+	return true
+}
+
+// AnyFrameworkVersionInClusterOutOfSync checks if any FrameworkVersion object matches a given Framework name and
+// if not it returns false. False means that for the given Framework the most recent official FrameworkVersion
+// is not installed in the cluster or an error occurred.
+func (k *K2oClient) FrameworkVersionInClusterOutOfSync(framework, mostRecentVersion string) bool {
+	fv, err := k.clientset.KudoV1alpha1().FrameworkVersions(vars.Namespace).List(v1.ListOptions{})
+	if err != nil {
+		return false
+	}
+	if len(fv.Items) < 1 {
+		return false
+	}
+
+	var i int
+	for _, v := range fv.Items {
+		if strings.HasPrefix(v.Name, framework) {
+			if v.Spec.Version == mostRecentVersion {
+				i++
+			}
+		}
+	}
+	if i < 1 {
+		return false
+	}
+	return true
+}
+
+// InstallFrameworkYamlToCluster expects a valid Framework obj to install
+func (k *K2oClient) InstallFrameworkYamlToCluster(obj *v1alpha1.Framework) (*v1alpha1.Framework, error) {
```

Comment:
You're not passing in YAML here, so maybe change the name of the function (and the next two)

Reply:
yes, I will change it similar as in this response https://github.com/kudobuilder/kudo/pull/138#discussion_r263937441